include/chopper/pack/hierarchical_binning.hpp

```diff
@@ -308,20 +308,17 @@ class hierarchical_binning
                 *data->output_buffer << "#FILES\tBIN_INDICES\tNUMBER_OF_BINS" << std::endl;
         }
 
-        size_t high_level_max_id{};
-        size_t high_level_max_size{};
+        bool const high = data->previous.empty(); // is this the top level ibf or not
 
-        size_t bin_id{};
+        // backtracking starts at the bottom right corner:
+        size_t trace_i = num_technical_bins - 1;
+        int trace_j = num_user_bins - 1;
```

Comment:
Why is this an `int`?
If it's used as parameter for `operator[]`, it should be `size_t`.

Reply:
Got it. A few lines further down I start the while loop

`while (trace_j >= 0)`

so at some point `trace_j` will be `-1` to end the loop. I'm sure it can be done differently with breaking the loop at the correct spot. I will check this in a follow up PR